client.go

```diff
@@ -1081,6 +1100,8 @@ func (m *Client) DeployContract(auth *bind.TransactOpts, name string, abi abi.AB
 				strings.Contains(strings.ToLower(err.Error()), "no contract code after deployment")
 		}),
 	); err != nil {
+		// do not pass the error here, because it's not transaction submission error
+		_, _ = m.Decode(tx, nil)
```

Comment:
Can we still use errors.Wrap()? Can it indicate both that we failed to wait for the deployed contract to appear on the chain and that we failed to decode the transaction?